emacs/radian.el

```diff
@@ -2568,7 +2568,12 @@ order."
     "Like `dumb-jump-go-prompt' but use a different window."
     (interactive)
     (let ((dumb-jump-window 'other))
-      (dumb-jump-go-prompt))))
+      (dumb-jump-go-prompt)))
+
+  (radian-defhook radian--dumb-jump-enable-xref-backend ()
+    xref-backend-functions
+    "Add `dumb-jump-xref-activate' hook to `xref-backend-functions'."
+    #'dumb-jump-xref-activate)))
```

Comment:
Does this do what you want? Shouldn't we be saying `(add-hook 'xref-backend-functions #'dumb-jump-xref-activate)`?
There's also the issue that we're actually running a sufficiently old version of `dumb-jump` that this function doesn't exist, so we should update the hash in the lockfile by means of `straight-freeze-versions`.

Reply:
If we're going to be upgrading `dumb-jump`, then it looks from the linked thread that we also need to install some compatibility hooks and possibly undeprecate some variables, at least until the conflict with `lsp-mode` gets resolved.

Reply:
I just started using Radian recently, so I'm still not very familiar with how it works. I thought the `radian-defhook` macro was the way to add the hook. But yes, seems like we do need the `(add-hook 'xref-backend-functions #'dumb-jump-xref-activate)` somewhere. I ended up putting it in the `radian-local-on-hook before-straight`(?)

> There's also the issue that we're actually running a sufficiently old version of dumb-jump that this function doesn't exist, so we should update the hash in the lockfile by means of straight-freeze-versions.

If we are locked in a specific version (I just noticed the lock file `versions.el`), then we all should either be getting or not getting the warnings, right? (I got them on the Radian install)

> If we're going to be upgrading dumb-jump, then it looks from the linked thread that we also need to install some compatibility hooks and possibly undeprecate some variables, at least until the conflict with lsp-mode gets resolved.

Yes, it's cumbersome because it seems that dumb-jump pulled the trigger on that xref implementation too soon without a complete fallback solution making the whole thing confusing.

But, as far as I understand (and I might not be getting this correctly), the `dumb-jump-xref-activate` hook adds `dumb-jump` to xref's list of backends so that when using one of its commands (e.g., `xref-find-definitions`) if it doesn't find anything fallback to dumb-jump's definitions (1 elisp -> 2 tags -> 3 dumb-jump).

(With the LSP problem that - and I haven't tested this out - it breaks that fallback lookup chain)

So I guess some of the options are:

1) Roll-back dumb-jump (? - why am I getting these warnings?)

2) Apply dump-jump's xref activation hook with a note to the README, and probably update [these bindings](https://github.com/raxod502/radian/blob/develop/emacs/radian.el#L2544-L2564) in favor of the preferred commands (e.g., `dumb-jump-go` -> `xref-find-definitions` since, in the end, it will still use dumb-jump if anything better is available) 

https://github.com/jacktasia/dumb-jump#obsolete-commands-and-options 

https://github.com/jacktasia/dumb-jump/issues/353#issuecomment-671195934

That should take care of the warnings and put Radian in what seems like the right direction - favoring xref while keeping dumb-jump.

3) Silence the warnings with `(put 'dumb-jump-go 'byte-obsolete-info
   nil)` until LSP issue has been resolved.

4) Leave it as is until LSP issue has been resolved

5) Consider how important dumb-jump is to Radian users and maybe remove it(?) If a user still cares enough for having it as xref's last resort, they can add/configure it in their `init.local.el`.

I'm probably good with options 2 or 5.

I also found these couple threads informative:

https://github.com/syl20bnr/spacemacs/pull/14206 

https://www.reddit.com/r/emacs/comments/hzxvke/how_do_people_have_dumbjump_setup/ 
(seems like there can also be an issue with the Tags backend)

Reply:
> I ended up putting it in the radian-local-on-hook before-straight

Personally, I would probably put it within `radian-local-on-hook after-init`, but either one will work; it is a matter of taste.

> If we are locked in a specific version (I just noticed the lock file versions.el), then we all should either be getting or not getting the warnings, right?

Yes, absolutely. If you are getting a different revision of dumb-jump than I am, then there is a bug in Radian or `straight.el`. Check `emacs/versions.el` and see if `~/.emacs.d/straight/repos/dumb-jump` has the correct version checked out. Did you perhaps forget to symlink `emacs/versions.el` to `~/.emacs.d/straight/versions/radian.el`? If so, then after doing that, you'd want to `M-x straight-thaw-versions` to reset the revisions of all checked-out packages to what is specified in the lockfile. If that's not the problem, then I'd suggest opening a new issue for that problem, and we can investigate further.

> Leave it as is until LSP issue has been resolved

I'd prefer to do this. It sounds like there is active discussion upstream on how to deal with this mess, and I'd prefer to avoid getting involved unless we have to. Hopefully, we can get away with not doing any specific work.

Reply:
I think you are right, at this time it's probably not worth it to try to fix this. I'll open an issue about the versioning problem, I did have the `versions.el` file linked.